### ISSUE TITLE
Turn fil-deal-ingester into service

### ISSUE THREAD

**Issue by juliangruber** [filecoin-station/fil-deal-ingester#23]
Instead of manually running the script, automate it in a service. This way we can also increase the ingestion frequency.

Add monitoring:
- [ ] no data point received
- [ ] an error happened during processing
- [ ] `added` below threshold
- [ ] `deleted` above threshold

**Comment by pyropy** [comment]
This could potentially be automated using [github actions](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#schedule).

**Comment by juliangruber** [comment]
Great idea!

**Comment by pyropy** [comment]
There seems to be a [limit on duration and storage](https://docs.github.com/en/billing/managing-billing-for-your-products/managing-billing-for-github-actions/about-billing-for-github-actions#included-storage-and-minutes) for Github runners. We could [self-host Github runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners) to mitigate this problem.

**Comment by pyropy** [comment]
> There seems to be a [limit on duration and storage](https://docs.github.com/en/billing/managing-billing-for-your-products/managing-billing-for-github-actions/about-billing-for-github-actions#included-storage-and-minutes) for Github runners (in our requirements we state that job between 3 - 5hrs and could take up to 40GB of space). We could [self-host Github runners](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners) to mitigate this problem.

Simple version could be just to run [scheduled machines on fly.io ](https://community.fly.io/t/new-feature-scheduled-machines/7398)

**Comment by juliangruber** [comment]
@pyropy and me discussed this and agreed that this would be nice to add to M4.12, since it's a quick win and will remove an annoying manual task

**Comment by bajtos** [comment]
It would be great if the new automated process could post a message to Slack with the summary. I think that's the very minimum we need for observability - if we see the message, then we know all is good; if we don't then we know we need to investigate.

```
**FINISHED INGESTION OF f05 DEALS**
Deleted: 258774
Added: 323142
Updated allocator info for 9 clients
```

**Comment by juliangruber** [comment]
Good idea to make this visible. Posting to Slack is one option, tracking in Influx, with Grafana alerts, is another, if we know how to set the alerts.

**Comment by bajtos** [comment]
This PR seems related: https://github.com/filecoin-station/fil-deal-ingester/pull/26

**Comment by bajtos** [comment]
I updated the docs in filecoin-project/service-classes here: https://github.com/filecoin-project/service-classes/commit/807675d42a0d610ab0d851b13f53e0b6bdbe94b9

**Comment by bajtos** [comment]
Loosely related: https://github.com/filecoin-station/fil-deal-ingester/issues/27

**Comment by pyropy** [comment]
Closing as we've automated both deal ingestion and reports.
